### PR TITLE
Ruff linter and formatter changes

### PR DESCRIPTION
**Pull request by gagandeepp**
Closes https://github.com/jpadilla/pyjwt/issues/996

1. Ruff linter and formatter settings implemented.
2. New test added to improve the code coverage.
3. Files formatted with ruff format.
4. Files checked with ruff check,

### CODE REVIEW

**Comment by gagandeepp** [comment]
@jpadilla  I have improved the code coverage, please review.
Somehow I was unable to open the previous MR on which you have commented.

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,65 @@
+[metadata]
```
Comment (jpadilla) [inline setup.cfg:1]: this required? they conflict with https://github.com/jpadilla/pyjwt/pull/995

**Review by jpadilla — CHANGES_REQUESTED** [review]

File: setup.cfg (hunk 1 above)
Comment (gagandeepp) [reply setup.cfg]: Removed

**Review by jpadilla — APPROVED** [review]
thanks!

**Comment by gagandeepp** [comment]
@jpadilla  Do you mind opening an issue to implement uv package manager in this repo?

**Pull request merged**